foursight_core/react/api/envs.py

```diff
@@ -0,0 +1,82 @@
+import copy
+from typing import Optional
+import os
+from dcicutils import ff_utils
+
+
+class Envs:
+
+    def __init__(self, known_envs: list):
+        # This known_envs should be the list of annotated environment name objects
+        # as returned by app_utils.get_unique_annotated_environment_names, where each
+        # object contains these fields: name, short_name, full_name, public_name, foursight_name
+        self.known_envs = known_envs
+
+    def get_known_envs(self) -> list:
+        return self.known_envs
+
+    def get_known_envs_with_gac_names(self, gac) -> list:
+        known_envs = copy.deepcopy(self.get_known_envs())
+        for known_env in known_envs:
+            known_env["gac_name"] = gac.get_gac_name(known_env["full_name"])
+        return known_envs
+
+    def get_default_env(self) -> str:
+        return os.environ.get("ENV_NAME", "no-default-env")
+
+    def is_known_env(self, env: str) -> bool:
+        return self.find_known_env(env) is not None
+
+    def find_known_env(self, env: str) -> Optional[dict]:
+        if not env:
+            return None
+        env = env.lower()
+        for known_env in self.known_envs:
+            if (known_env["name"].lower() == env
+                or known_env["short_name"].lower() == env
+                or known_env["full_name"].lower() == env
```

Comment:
Why are these not using the ordinary `dcicutils.env_utils` functions? The whole ecosystem concept in `EnvUtils` is designed to make Foursight be able to call these regardless of which env is current so you can do operations like this. I would expect something like (untested):
```
from dcicutils.env_utils import foursight_env_name
from dcicutils.misc_utils import find_association
def find_known_env(self, env: str) -> Optional[dict]:
    return find_association(known_envs, foursight_name=foursight_env_name(env))
```
if you even need the known envs to be a dictionary. If it's just a list of names because you're using `env_utils` functions to get such info, then (untested):
```
from dcicutils.env_utils import foursight_env_name
def find_known_env(self, env: str, canonicalize=True) -> Optional[str]:
    sought_env = foursight_env_name(env)
    for known_env in known_envs:
        if foursight_env_name(known_env) == sought_env:
            return sought_env if canonicalize else env
    return None  # not found
```

Reply:
Ah, yes, I should (re)visit env_utils to see what's covered and what  I should use. I wasn't sure the functions there always take all versions of the name (short, full, public, foursight) into account when (e.g.) comparing, etc.

Reply:
Okay I think I've taken care of these, thanks.